### DISCUSSION TITLE
Running Teamcity Agent on USS with JAVA 11

### DISCUSSION THREAD

**Discussion by dennis-behm** [IBM/dbb#214] [Q&A]
Switching from JAVA 8 to JAVA 11 for the Teamcity agent running on Unix System Services causes some unforeseen issues around encoding and let pipelines fail:

* Cloning fails
* Shell tasks produce garbage output

Please make sure to update the settings from `teamcity.runner.commandline.script.encoding=ISO8859-1` to `teamcity.runner.commandline.script.encoding=IBM1047`.

They are set in the conf/buildAgent.properties file.

Additionally, depending on the selected authentication method to clone the repository, you might need to leverage a wrapper script like in [Utilities/Jenkins](https://github.com/IBM/dbb/tree/ddced2032bbb79824d055381adf947617e05d0f5/Utilities/Jenkins)  
that uses iconv to convert credentials to IBM-1047. 

Kudos to Jesper for sharing the results of his investigations for this problem.

**Comment by prab99** [comment]
i am seeing another anomaly between Java 8 and 11 with Teamcity agent (version 129421) running under zOS 2.5. Teamcity agentTmp does not have Tagging ON.
ls -lTr
Under Java 11
- untagged    T=off -rwxr-xr-x   1 TEAMCITY IBMDBB       639 Jun 12 17:30 custom_script18311974430943418313   

and  Under Java 8
t ISO8859-1   T=on  -rwxr-xr-x   1 TEAMCITY IBMDBB       639 Jun 12 17:36 custom_script8812846587072450007 

Causing EDC5130I Exec format error. I have even tried Mounting Teamcity ZFS files with CCSID with CCSID 1047 and 819(ISO8859) to Force Tagging , still the script fails. 

Do not have any solutions ATM.
